raiden/network/proxies/secret_registry.py

```diff
@@ -134,43 +131,34 @@ def register_secret_batch(self, secrets: List[Secret]) -> None:
         # From here on the lock is not required. Context-switches will happen
         # for the gas estimation and the transaction, however the
         # synchronization data is limited to the open_secret_transactions
-        log_details = {
-            "node": to_checksum_address(self.node_address),
-            "contract": to_checksum_address(self.address),
-            "secrethashes": secrethashes_to_register,
-            "secrethashes_not_sent": secrethashes_not_sent,
-        }
+        if secrets_to_register:
+            log_details = {"secrethashes_not_sent": secrethashes_not_sent}
+            self._register_secret_batch(secrets_to_register, transaction_result, log_details)
 
-        with log_transaction(log, "register_secret_batch", log_details):
-            if secrets_to_register:
-                self._register_secret_batch(secrets_to_register, transaction_result, log_details)
-
-            gevent.joinall(wait_for, raise_error=True)
+        gevent.joinall(wait_for, raise_error=True)
 
     def _register_secret_batch(
         self,
         secrets_to_register: List[Secret],
         transaction_result: AsyncResult,
-        log_details: Dict[Any, Any],
+        log_details: Dict[str, Any],
     ) -> None:
-        checking_block = self.client.get_checking_block()
-        gas_limit = self.client.estimate_gas(
-            self.proxy, checking_block, "registerSecretBatch", secrets_to_register
+
+        estimated_transaction = self.client.estimate_gas(
+            self.proxy, "registerSecretBatch", log_details, secrets_to_register
         )
         receipt = None
         transaction_hash = None
         msg = None
 
-        if gas_limit:
-            gas_limit = safe_gas_limit(
-                gas_limit, len(secrets_to_register) * GAS_REQUIRED_PER_SECRET_IN_BATCH
+        if estimated_transaction is not None:
+            estimated_transaction.estimated_gas = safe_gas_limit(
+                estimated_transaction.estimated_gas,
+                len(secrets_to_register) * GAS_REQUIRED_PER_SECRET_IN_BATCH,
             )
-            log_details["gas_limit"] = gas_limit
 
             try:
-                transaction_hash = self.client.transact(
-                    self.proxy, "registerSecretBatch", gas_limit, secrets_to_register
```

Comment:
the arguments don't have to be duplicated anymore, they are part of the `transaction` object, created by `self.client.estimate_gas`